primitives/arithmetic/src/helpers_128bit.rs

```diff
@@ -191,9 +205,6 @@ pub const fn multiply_by_rational_with_rounding(
 	r: Rounding,
 ) -> Option<u128> {
 	use double128::Double128;
-	if c == 0 {
-		return None
-	}
```

Comment:
This is actually making it less secure, which is why i would rather put the logic in the checked variant (basically same comment as above).